src/pykeen/models/nmodel/base.py

```diff
@@ -0,0 +1,547 @@
+            The indices of (head, relation, tail) triples.
+
+        :return: shape: (batch_size, 1), dtype: float
+            The score for each triple.
+        """
+        return self(
+            h_indices=hrt_batch[:, 0],
+            r_indices=hrt_batch[:, 1],
+            t_indices=hrt_batch[:, 2],
+        ).view(hrt_batch.shape[0], 1)
+
+    def score_t(self, hr_batch: torch.LongTensor, slice_size: Optional[int] = None) -> torch.FloatTensor:
+        """Forward pass using right side (tail) prediction.
+
+        This method calculates the score for all possible tails for each (head, relation) pair.
+
+        :param hr_batch: shape: (batch_size, 2), dtype: long
+            The indices of (head, relation) pairs.
+        :param slice_size:
+            The slice size.
+
+        :return: shape: (batch_size, num_entities), dtype: float
+            For each h-r pair, the scores for all possible tails.
+        """
+        return self(
+            h_indices=hr_batch[:, 0],
+            r_indices=hr_batch[:, 1],
+            t_indices=None,
+            slice_size=slice_size,
+            slice_dim="h",
+        ).view(hr_batch.shape[0], self.num_entities)
+
+    def score_h(self, rt_batch: torch.LongTensor, slice_size: Optional[int] = None) -> torch.FloatTensor:
+        """Forward pass using left side (head) prediction.
+
+        This method calculates the score for all possible heads for each (relation, tail) pair.
+
+        :param rt_batch: shape: (batch_size, 2), dtype: long
+            The indices of (relation, tail) pairs.
+        :param slice_size:
+            The slice size.
+
+        :return: shape: (batch_size, num_entities), dtype: float
+            For each r-t pair, the scores for all possible heads.
+        """
+        return self(
+            h_indices=None,
+            r_indices=rt_batch[:, 0],
+            t_indices=rt_batch[:, 1],
+            slice_size=slice_size,
+            slice_dim="r",
+        ).view(rt_batch.shape[0], self.num_entities)
+
+    def score_r(self, ht_batch: torch.LongTensor, slice_size: Optional[int] = None) -> torch.FloatTensor:
+        """Forward pass using middle (relation) prediction.
+
+        This method calculates the score for all possible relations for each (head, tail) pair.
+
+        :param ht_batch: shape: (batch_size, 2), dtype: long
+            The indices of (head, tail) pairs.
+        :param slice_size:
+            The slice size.
+
+        :return: shape: (batch_size, num_relations), dtype: float
+            For each h-t pair, the scores for all possible relations.
+        """
+        return self(
+            h_indices=ht_batch[:, 0],
+            r_indices=None,
+            t_indices=ht_batch[:, 1],
+            slice_size=slice_size,
+            slice_dim="t",
+        ).view(ht_batch.shape[0], self.num_relations)
+
+
+def collect_regularization_terms(module: nn.Module) -> Union[float, torch.FloatTensor]:
```

Comment:
This won't work because the regularization class hasn't been updated. I guess we have to take the entire new set of regularizers over for this

Reply:
@cthoyt this comment is fixed by #266 ?

Reply:
not yet. i'm actually looking at this code ATM